keras/engine/training.py

```diff
@@ -1893,9 +1893,11 @@ def fit_generator(self, generator,
                 non picklable arguments to the generator
                 as they can't be passed
                 easily to children processes.
-            shuffle: Whether to shuffle the data at the beginning of each
-                epoch. Only used with instances of `Sequence` (
-                keras.utils.Sequence).
+            shuffle: Whether to shuffle the order of the batches at 
+                the beginning of each epoch. Only used with instances 
+                of `Sequence` (keras.utils.Sequence). To shuffle the
+                data itself, your `Sequence` should implement 
+                `on_epoch_end`.
```

Comment:
That last sentence will be very difficult to understand for people who don't have further context.

Reply:
Thank you for the review & feedback.

The problem is that the `shuffle` option here is misleading. I expected it to behave similarly to the `shuffle` option in `model.fit(...)`, but of course it can't, because it doesn't have access to shuffle the generator's underlying data.  I was trying to improve the situation with documentation.  

Another idea might be to change it `shuffle` from accepting `True` or `False` to accepting `"batch"` or `None`, which I believe is similar in usage & meaning to the `shuffle` option in `model.fit(...)`.  

Or I can make another attempt at updating the documentation, if you think that's a better solution.  Or we can close this and forget about it!

Reply:
"batch shuffling" has a different meaning (shuffling data within mini-batches, but not globally). I think the change in docstring you propose is fine as long as the last sentence is removed.